shor/transpilers/qiskit.py

```diff
@@ -6,6 +6,7 @@
 
 
 def to_qiskit_circuit(shor_circuit: QC) -> QuantumCircuit:
+    # shor_registers = shor_circuit.to_registers()  # unsure here how to add registers like QuantumCircuit(3,1,1)
     qiskit_circuit = QuantumCircuit(
         len(shor_circuit.initial_state()),
         len(shor_circuit.measure_bits()),
```

Comment:
This is the line you will want to change.

Currently this relies on the initial state to determine the Qbit register size, and the measure_bits() to determine the classical register size.

But since these should be in the circuit, you should be able to set them explicitly.

I see that you are trying to do this in a function (to registers) and that is good idea.

Implementation looks a little off. Commenting above.

Reply:
I was thinking about just len(qubit_registers) and len(cbit_registers) but I believe qiskit makes a different circuit for QuantumCircuit(3,2) vs QuantumCircuit(3,1,1) or at least a different circuit pops up in their diagram